### PR TITLE
docs: add BUILDING.md

### PR DESCRIPTION
**Pull request by jpculp**
**Issue number:**
#57 

**Description of changes:**
Adds `BUILDING.md`.

Also adds availability section to `README.md` for published images.

**Testing done:**
Previewed in `vscode`.

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

File: BUILDING.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+## Build the images
+
+### Dependencies
+
+#### System Requirements
+
+The build process images can consume in excess of 40GB in the docker directory.
+
+The build process is also fairly demanding on your CPU, since we build all included software from scratch.
```
Comment (webern) [inline BUILDING.md]: nit

```suggestion
The build process is also fairly demanding on your CPU since we build all included software from scratch.
```

File: BUILDING.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
+# Building the Bottlerocket SDK
+
+If you'd like to build your own images instead of relying on those available on ECR Public, follow these steps.
```
Comment (webern) [inline BUILDING.md]: I think we should point to where they are in the README.

Note, however. that the README doesn't say where they are. I think the README should be updated to include a link to the ECR Public landing page for the Bottlerocket SDK. I recommend adding a commit to the this PR adding such a link to the README.

```suggestion
If you'd like to build your own images instead of relying on the published images, follow these steps.
```

File: BUILDING.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+To build the images, run:
+
+```
+make ARCH=my-target-arch-here
+```
+
+One thing to keep in mind is the difference between the container host architecture and target architecture (the `ARCH` argument being for target arch).
+If you plan on building for each architecture, you will want to build each target architecture on each host architecture.
+In other words, you’ll want to do `make ARCH=x86_64` and `make ARCH=aarch64` on an **x86_64** machine and an **aarch64** machine for a total of four builds.
```
Comment (webern) [inline BUILDING.md]: This edit reads a little more clearly to me.

```suggestion
In other words, you’ll want to do `make ARCH=x86_64` and `make ARCH=aarch64` on an **x86_64** machine and again on an **aarch64** machine for a total of four builds.
```

File: BUILDING.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+From there, you will be able to [build Bottlerocket](https://github.com/bottlerocket-os/bottlerocket/blob/develop/BUILDING.md).
+
+## Publish your images
+
+To publish your images, you will have to create repositories for the SDK and toolchain images, once again specifying the target architecture:
+
+- my-custom-bottlerocket-sdk-aarch64
+- my-custom-bottlerocket-sdk-x86_64
+- my-custom-bottlerocket-toolchain-aarch64
+- my-custom-bottlerocket-toolchain-x86_64
```
Comment (webern) [inline BUILDING.md]: Lines 77 and 78 are good, but 70-75 might be untrue. I'm assuming you could put them into a single repository if you so choose using different tags. Does Makefile.toml assume this repository naming structure? If so it might help to clarify the reason for the naming convention and repo structure.

Comment (jpculp) [reply BUILDING.md]: The repository structure above is what we use today. You could certainly change it up, but I think for the purposes of this guide it should be closer to how we publish our own images.

Comment (webern) [reply BUILDING.md]: > you will have to create repositories for the SDK and toolchain images

Why?

I suggest rewording it somehow. When we publish pushing code then it will make more sense I suppose but for now I'm just wondering what is going to break if I don't do this since you said I *have to*.

**Comment by jpculp** [comment]
Addressed @webern's comments and added a link to the `BUILDING.md` in the `README.md`.

File: README.md [hunk 5]
```diff
@@ -13,6 +13,15 @@ The SDK includes:
 * Toolchains for Rust and Go
 * Software license scanner
 
+### Availability
+
+The SDK is available through Amazon ECR Public Gallery:
```
Comment (webern) [inline README.md:18]: I think the gallery is just the subdomain for the web browsing UI.

```suggestion
The SDK is available through Amazon ECR Public:
```

**Review by webern — APPROVED** [review]

File: README.md (hunk 5 above)
Comment (jpculp) [reply README.md:18]: It's a little bit more than just a subdomain name. The registry itself is 'Amazon ECR Public', whereas the website is 'Amazon ECR Public Gallery'.

**Review by webern — APPROVED** [review]

**Comment by jpculp** [comment]
- Added information about publish target and script to **BUILDING.md**.
- Slightly improved readability and wording.

**Comment by jpculp** [comment]
- Moved **README.md** tweaks from the 2nd commit to the 1st.

**Review by cbgbt — APPROVED** [review]

**Review by bcressey — APPROVED** [review]
